src/dyn_styles.rs

```diff
@@ -0,0 +1,505 @@
+        /// Change the background color to yellow
+        Yellow   yellow   on_yellow,
+        /// Change the foreground color to blue
+        /// Change the background color to blue
+        Blue     blue     on_blue,
+        /// Change the foreground color to magenta
+        /// Change the background color to magenta
+        Magenta  magenta  on_magenta,
+        /// Change the foreground color to purple
+        /// Change the background color to purple
+        Magenta  purple   on_purple,
+        /// Change the foreground color to cyan
+        /// Change the background color to cyan
+        Cyan     cyan     on_cyan,
+        /// Change the foreground color to white
+        /// Change the background color to white
+        White    white    on_white,
+
+        /// Change the foreground color to bright black 
+        /// Change the background color to bright black
+        BrightBlack    bright_black    on_bright_black,
+        /// Change the foreground color to bright red
+        /// Change the background color to bright red
+        BrightRed      bright_red      on_bright_red,
+        /// Change the foreground color to bright green
+        /// Change the background color to bright green
+        BrightGreen    bright_green    on_bright_green,
+        /// Change the foreground color to bright yellow
+        /// Change the background color to bright yellow
+        BrightYellow   bright_yellow   on_bright_yellow,
+        /// Change the foreground color to bright blue
+        /// Change the background color to bright blue
+        BrightBlue     bright_blue     on_bright_blue,
+        /// Change the foreground color to bright magenta
+        /// Change the background color to bright magenta
+        BrightMagenta  bright_magenta  on_bright_magenta,
+        /// Change the foreground color to bright purple
+        /// Change the background color to bright purple
+        BrightMagenta  bright_purple   on_bright_purple,
+        /// Change the foreground color to bright cyan
+        /// Change the background color to bright cyan
+        BrightCyan     bright_cyan     on_bright_cyan,
+        /// Change the foreground color to bright white
+        /// Change the background color to bright white
+        BrightWhite    bright_white    on_bright_white,
+    }
+
+    style_methods! {
+        /// Make the text bold
+        bold BOLD,
+        /// Make the text dim
+        dimmed DIMMED,
+        /// Make the text italicized
+        italic ITALIC,
+        /// Make the text italicized
+        underline UNDERLINED,
+        /// Make the text blink
+        blink BLINK,
+        /// Make the text blink (but fast!)
+        blink_fast BLINK_FAST,
+        /// Swap the foreground and background colors
+        reversed REVERSED,
+        /// Hide the text
+        hidden HIDDEN,
+        /// Cross out the text
+        strikethrough STRIKETHROUGH,
+    }
+
+    #[inline(always)]
```

Comment:
Have you benchmarked all these inlinings? All the other inlines were based on running a lot of benchmarks, I'm not very confident in some of these, especially the larger ones.

Reply:
No, I did not benchmark this. I just didn't delete it when I copied your code over to adapt it.

Do you still have the benchmark code lying around, so I can adapt it?

Or should I just delete the attributes?

Reply:
I'd say delete the attributes for now. I'm afraid I don't really have the benchmarks.